account-kit/smart-contracts/src/ma-v2/client/client.ts

```diff
@@ -25,13 +35,34 @@ export type CreateSMAV2AccountClientParams<
     SmartAccountClientConfig<TTransport, TChain>,
     "transport" | "account" | "chain"
   >;
+export type CreateSMAV2AlchemyAccountClientParams<
+  TTransport extends Transport = Transport,
+  TChain extends Chain = Chain,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = Omit<
+  CreateSMAV2AccountClientParams<TTransport, TChain, TSigner>,
+  "transport"
+> &
+  Omit<
+    AlchemySmartAccountClientConfig<TChain, LightAccount<TSigner>>,
+    "account"
+  > & { paymasterAndData?: never };
 
 export function createSMAV2AccountClient<
   TChain extends Chain = Chain,
   TSigner extends SmartAccountSigner = SmartAccountSigner
 >(
-  args: CreateSMAV2AccountClientParams<Transport, TChain, TSigner>
-): Promise<SMAV2AccountClient<TSigner>>;
+  args: CreateSMAV2AlchemyAccountClientParams<AlchemyTransport, TChain, TSigner>
+): Promise<SMAV2AccountClient<TSigner, TChain, AlchemyTransport>>;
+
+export function createSMAV2AccountClient<
+  TTransport extends Transport = Transport,
+  TChain extends Chain = Chain,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+>(
+  args: CreateSMAV2AccountClientParams<TTransport, TChain, TSigner> &
+    NotType<TTransport, AlchemyTransport>
```

Comment:
📓 So, this here was needed for the test to make sure that we didn't filter on the type alchemyTransport for the other case